### PR TITLE
Project 14

### PR DESCRIPTION
**Pull request by volodymyrtaras**

### CODE REVIEW

File: src/main/java/fourteen/absfactory/CodeApp.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package fourteen.absfactory;
+
+public class CodeApp {
+    public static void main(String[] args) {
+        Engine engine = configure();
+        engine.notification();
+    }
+
+    private static Engine configure() {
```
Comment (Taniann) [inline src/main/java/fourteen/absfactory/CodeApp.java]: better move it to the saparate class
dont mix class with method name with logic methods

File: src/main/java/fourteen/absfactory/CodeApp.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        Engine engine;
+        String osName = System.getProperty("os.name").toLowerCase();
+        if (osName.contains("linux")) {
```
Comment (Taniann) [inline src/main/java/fourteen/absfactory/CodeApp.java]: use constants for "linux" string

File: src/main/java/fourteen/adapter/MicroUsb.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package fourteen.adapter;
+
+public class MicroUsb {
+    private static final String CONNECTION_STATUS = "Connection via MicroUsb was created!";
+    public String getConnection() {
```
Comment (Taniann) [inline src/main/java/fourteen/adapter/MicroUsb.java]: add blank line between fields and methods

File: src/main/java/fourteen/flyweight/HumanFactory.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.Map;
+
+public class HumanFactory {
+    private static final Map<String, Human> MAP = new HashMap<>();
+
+    public Human getHuman(String name) {
+        Human human = MAP.get(name);
+        if (human == null) {
+            switch (name) {
+                case "Man":
```
Comment (Taniann) [inline src/main/java/fourteen/flyweight/HumanFactory.java]: better use constants

File: src/main/java/fourteen/factory/PopMusic.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package fourteen.factory;
+
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Builder
```
Comment (Taniann) [inline src/main/java/fourteen/factory/PopMusic.java]: craftily :)

**Comment by volodymyrtaras** [comment]
@Taniann fixed.

**Pull request merged**